assets/js/components/Dashboard/NewPaMessage/SelectedStationsSummary.tsx

```diff
@@ -0,0 +1,144 @@
+import { Place } from "Models/place";
+import { Screen } from "Models/screen";
+import React, { useEffect, useState } from "react";
+import { GeoAltFill, X } from "react-bootstrap-icons";
+import fp from "lodash/fp";
+import { SILVER_LINE_ROUTES } from "Constants/constants";
+import { classWithModifier } from "../../../util";
+import pluralize from "pluralize";
+interface Props {
+  value: string[];
+  onChange: (zones: string[]) => void;
+  places: Place[];
+  busRoutes: string[];
+}
+
+const SelectedGroupTag = ({
+  numPlaces,
+  routeId,
+  onClick,
+}: {
+  numPlaces: number;
+  routeId: string;
+  onClick: () => void;
+}) => {
+  if (numPlaces === 0) return null;
+  return (
+    <div
+      className={classWithModifier("selected-group-tag", routeId.toLowerCase())}
+    >
+      {routeId}: {numPlaces} {pluralize("Station", numPlaces)}
+      <button className={routeId.toLowerCase()} onClick={onClick}>
+        <X />
+      </button>
+    </div>
+  );
+};
+
+const SelectedStationsSummary = ({
+  value,
+  places,
+  busRoutes,
+  onChange,
+}: Props) => {
+  const [placesWithSelectedScreens, setPlacesWithSelectedScreens] = useState<
+    Place[]
+  >([]);
+  useEffect(() => {
+    const placesWithSelectedScreens = fp.flow(
+      fp.map((place: Place) => {
+        return {
+          ...place,
+          screens: place.screens.filter((screen) =>
+            fp.includes(screen.id, value),
+          ),
+        };
+      }),
+      fp.filter((place) => place.screens.length > 0),
+    )(places);
+
+    setPlacesWithSelectedScreens(placesWithSelectedScreens);
+  }, [value]);
```

Comment:
**suggestion:** a `useEffect` call which just computes a value and calls some `setThing` function can usually be replaced with a single `useMemo`

```typescript
  const placesWithSelectedScreens = useMemo<Place[]>(() => {
    return fp.flow(
      fp.map((place: Place) => {
        return {
          ...place,
          screens: place.screens.filter((screen) =>
            fp.includes(screen.id, value),
          ),
        };
      }),
      fp.filter((place) => place.screens.length > 0),
    )(places);
  }, [value]);
```
(formatting probably needs some help)

Reply:
O sweet thank you.